examples/component-provider/randomComponent.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016-2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	p "github.com/pulumi/pulumi-go-provider"
+	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type RandomComponent struct {
+	pulumi.ResourceState
+	RandomComponentArgs
+	Password        pulumi.StringOutput `pulumi:"password"`
+	HardcodedOutput pulumi.StringOutput `pulumi:"hardcodedOutput"`
+}
+
+type RandomComponentArgs struct {
+	Length pulumi.IntInput `pulumi:"length"`
+}
+
+func NewMyComponent(ctx *pulumi.Context, name string, compArgs RandomComponentArgs, opts ...pulumi.ResourceOption) (*RandomComponent, error) {
```

Comment:
Were you planning to actually use `NewNestedRandomComponent` in here? Where are you demonstrating nesting?

Reply:
This is to demonstrate/test the ability that we can use code that is in a different module than one that lives in main. Perhaps the name may need to be clearer, but the current example is not to test nested components, but rather, nested modules.

 Eg. https://github.com/pulumi/pulumi-go-provider/pull/317/files/472ab11c0fb6bb78b60794776a5e642de869e5d4#diff-0a58ad0bb7ae01516ba413795b3cd9e0a0d651f3ce4dff46bbdb1e33c7fc20f5